### PR TITLE
Don't delete the whole destination folder while saving entityset

### LINKED ISSUE #706: Saving entityset deletes whole destination folder
https://github.com/Featuretools/featuretools/blob/97eb274a48752985cd33d62a92eef2a16c874d5b/featuretools/entityset/serialize.py#L148

In the current implementation, when an entity set is serialized, the user must give a target folder for the entityset to go in. However, when Featuretools serializes the entityset, it deletes the whole folder if it exists and then copies the serialized data set into it. Instead, featuretools should only delete conflicting files (files that are the same as the ones it is trying to put in), and leave the rest of the files alone.

### PR DESCRIPTION
**Pull request by chidauri**
### Pull Request Description
Fixes #706  as the title suggests. The new functionality is to replace the conflicting files only and not the whole path.

-----
*After creating the pull request: in order to pass the **changelog_updated** check you will need to update the "Future Release" section of* `docs/source/changelog.rst` *to include this pull request.*

### CODE REVIEW

**Comment by rwedge** [comment]
Thanks again, @chidauri.  Could you add a test case confirming that this no longer deletes unrelated files in the destination folder?

File: featuretools/tests/entityset_tests/test_serialization.py [hunk 1]
```diff
@@ -268,6 +268,13 @@ def test_serialize_url_csv(es):
         es.to_csv(URL, encoding='utf-8', engine='python')
 
 
+def test_serialize_data_description(es, tmpdir):
```
Comment (rwedge) [inline featuretools/tests/entityset_tests/test_serialization.py]: this test name could be updated to provide more context about that it is testing extra files are not removed

**Comment by rwedge** [comment]
@chidauri thanks for adding the test case!  I think it would also be good to test that old serialization files (e.g. data_description.json) get overwritten.

File: featuretools/tests/entityset_tests/test_serialization.py (hunk 1 above)
Comment (chidauri) [reply featuretools/tests/entityset_tests/test_serialization.py]: Is it okay now?

Comment (rwedge) [reply featuretools/tests/entityset_tests/test_serialization.py]: Looks like there's an issue with py35 / py27 tests.

Comment (chidauri) [reply featuretools/tests/entityset_tests/test_serialization.py]: Fixed :)

**Pull request merged**